system-monitor@paradoxxx.zero.gmail.com/extension.js

```diff
@@ -635,52 +627,47 @@ const Pie = new Lang.Class({
         if (Compat.versionCompare(shell_Version, '3.7.4')) {
             cr.$dispose();
         }
-    },
-    update_mounts: function (mounts) {
+    }
+
+    update_mounts(mounts) {
         this.mounts = mounts;
         this.actor.queue_repaint();
     }
-});
-
-const TipItem = new Lang.Class({
-    Name: 'SystemMonitor.TipItem',
-    Extends: PopupMenu.PopupBaseMenuItem,
+}
 
-    _init: function () {
-        PopupMenu.PopupBaseMenuItem.prototype._init.call(this);
+const TipItem = class SystemMonitor_TipItem extends PopupMenu.PopupBaseMenuItem {
+    constructor() {
+        super();
+        // PopupMenu.PopupBaseMenuItem.prototype._init.call(this);
         this.actor.remove_style_class_name('popup-menu-item');
         this.actor.add_style_class_name('sm-tooltip-item');
     }
-});
-
-const TipMenu = new Lang.Class({
-    Name: 'SystemMonitor.TipMenu',
-    Extends: PopupMenu.PopupMenuBase,
+}
 
-    _init: function (sourceActor) {
+const TipMenu = class SystemMonitor_TipMenu extends PopupMenu.PopupMenuBase {
+    constructor(sourceActor) {
         // PopupMenu.PopupMenuBase.prototype._init.call(this, sourceActor, 'sm-tooltip-box');
-        this.parent(sourceActor, 'sm-tooltip-box');
-        this.actor = new Shell.GenericContainer();
-        this.actor.connect('get-preferred-width',
-            Lang.bind(this, this._boxGetPreferredWidth));
-        this.actor.connect('get-preferred-height',
-            Lang.bind(this, this._boxGetPreferredHeight));
-        this.actor.connect('allocate', Lang.bind(this, this._boxAllocate));
+        super(sourceActor, 'sm-tooltip-box');
+        this.actor = new Clutter.Actor();
+        // this.actor.connect('get-preferred-width',
+        //     this._boxGetPreferredWidth).bind(this);
+        // this.actor.connect('get-preferred-height',
+        //     this._boxGetPreferredHeight.bind(this));
         this.actor.add_actor(this.box);
-    },
-    _boxGetPreferredWidth: function (actor, forHeight, alloc) {
-        // let columnWidths = this.getColumnWidths();
-        // this.setColumnWidths(columnWidths);
-
-        [alloc.min_size, alloc.natural_size] = this.box.get_preferred_width(forHeight);
-    },
-    _boxGetPreferredHeight: function (actor, forWidth, alloc) {
-        [alloc.min_size, alloc.natural_size] = this.box.get_preferred_height(forWidth);
-    },
-    _boxAllocate: function (actor, box, flags) {
-        this.box.allocate(box, flags);
-    },
-    _shift: function () {
+    }
+    // _boxGetPreferredWidth (actor, forHeight, alloc) {
```

Comment:
Just wondering if it is correct to just comment these methods out... [The MR to deprecate Shell.GenericContainer](https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/153) seems to suggest we can do vfunc overriding now, and [this is what I tried](https://github.com/DreaminginCodeZH/gnome-shell-system-monitor-applet/commit/19467c3e464d6b7f75989e9b1d1087e91adddbd1#diff-3d6a322a2008ae119622ecc9bde5555cR674) according to the changes in it. I didn't know if this actually preserved something, but everything also worked fine in my fork. Just hope this might be of some help :)